src/activity_actor.cpp

```diff
@@ -1076,10 +1079,20 @@ void lockpick_activity_actor::finish( player_activity &act, Character &who )
     /** @EFFECT_DEX improves chances of successfully picking door lock, reduces chances of bad outcomes */
     /** @EFFECT_MECHANICS improves chances of successfully picking door lock, reduces chances of bad outcomes */
     /** @EFFECT_LOCKPICK greatly improves chances of successfully picking door lock, reduces chances of bad outcomes */
-    int pick_roll = std::pow( 1.5, who.get_skill_level( skill_traps ) ) *
-                    ( std::pow( 1.3, who.get_skill_level( skill_mechanics ) ) +
-                      it->get_quality( qual_LOCKPICK ) - it->damage() / 2000.0 ) +
-                    who.dex_cur / 4.0;
+    int pick_roll = ( std::pow( 1.5, who.get_skill_level( skill_traps ) ) *
```

Comment:
I think pick_roll needs to be a float throughout this process, otherwise it's going to have some pretty bad rounding behavior.

Reply:
yknow what, I'm just going to change the dang thing to use a normal_roll like I have done with traps... my code there has all the float stuff worked in.